consensus/scp/src/slot.rs

```diff
@@ -1394,18 +1386,25 @@ impl<V: Value, ValidationError: Display> Slot<V, ValidationError> {
         };
 
         // Return the union
-        let mut accepted_nominated: BTreeSet<V> = Default::default();
-        accepted_nominated.append(&mut accepted_from_blocking_set);
-        accepted_nominated.append(&mut accepted_from_quorum);
+        let mut accepted_nominated: HashSet<V> = Default::default();
+        accepted_nominated.extend(accepted_from_blocking_set.into_iter());
+        accepted_nominated.extend(accepted_from_quorum.into_iter());
 
         accepted_nominated
     }
 
     /// "Confirmed Nominated" values that are not yet in self.Z.
-    fn additional_values_confirmed_nominated(&self) -> BTreeSet<V> {
+    fn additional_values_confirmed_nominated(&self) -> HashSet<V> {
+        let mut initial_values: BTreeSet<V> = Default::default();
```

Comment:
These predicates still use BTreeSets. I don't see any reason why they need to be BTreeSets (instead of hashSets), but changing them to HashSet breaks some of the crate's integration tests. MC-1444 is about investigating this.